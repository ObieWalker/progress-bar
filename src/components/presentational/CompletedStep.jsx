import React from "react";

import {
  Text,
  Progress,
  CompleteStep,
  theme
} from '../../styles/stylesComponent';

const CompletedStep = (props) => {
  const { currentStep, stepNumber } = props
  const completed = currentStep >= stepNumber ? 'filled': ''

  return (
    <>
      <Progress completed={completed} theme={theme}></Progress>
      <CompleteStep completed={completed} theme={theme}>
      <Text>Step {props.stepNumber}</Text>
      </CompleteStep>
    </>
  )
}

export default CompletedStep;