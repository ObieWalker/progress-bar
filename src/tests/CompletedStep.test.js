import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import CompletedStep from '../../src/components/presentational/CompletedStep';

let mountedForm, props, wrapper;
const currentStep = 5
const stepNumber = 4

const getComponent = () => {
  if (!mountedForm) {
    props = {
      currentStep,
      stepNumber 
    };
    mountedForm = shallow(<CompletedStep {...props} />);
  }
  return mountedForm;
};

describe('ProgressBar Component', () => {
  beforeAll(() => {
    wrapper = getComponent();
  });
  it('renders component successfully', () => {
    expect(wrapper).toMatchSnapshot();
  });
});