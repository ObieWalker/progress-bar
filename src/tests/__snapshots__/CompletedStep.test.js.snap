// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar Component renders component successfully 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CompletedStep
    currentStep={5}
    stepNumber={4}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef
          completed="filled"
          theme={
            Object {
              "color": "rgb(58, 67, 141)",
            }
          }
        />,
        <ForwardRef
          completed="filled"
          theme={
            Object {
              "color": "rgb(58, 67, 141)",
            }
          }
        >
          <ForwardRef>
            Step 
            4
          </ForwardRef>
        </ForwardRef>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "completed": "filled",
          "theme": Object {
            "color": "rgb(58, 67, 141)",
          },
        },
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-EHOje",
            "isStatic": false,
            "rules": Array [
              "
  margin: 14px -5px 0 -5px;
  position: relative;
  height: 6px;
  width: 200px;
  border: 0.1em solid ",
              [Function],
              ";
  z-index: 0;
  ",
              [Function],
              ";
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-EHOje",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <ForwardRef>
            Step 
            4
          </ForwardRef>,
          "completed": "filled",
          "theme": Object {
            "color": "rgb(58, 67, 141)",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              "Step ",
              4,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Step ",
            4,
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-htoDjs",
              "isStatic": true,
              "rules": Array [
                "
  font-size: 15px;
  margin-top: 40px;
  text-align: center;
  font-family: 'Courier New', Courier, monospace;
",
              ],
            },
            "displayName": "styled.p",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-htoDjs",
            "target": "p",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bZQynM",
            "isStatic": false,
            "rules": Array [
              "
  font-size: 60px;
  display: flex;
  width: 0.4em;
  box-sizing: content-box;
  height: 0.4em;
  border: 0.1em solid ",
              [Function],
              ";
  position: relative;
  border-radius: 0.35em;
  z-index: 1
  ",
              [Function],
              ";
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bZQynM",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef
            completed="filled"
            theme={
              Object {
                "color": "rgb(58, 67, 141)",
              }
            }
          />,
          <ForwardRef
            completed="filled"
            theme={
              Object {
                "color": "rgb(58, 67, 141)",
              }
            }
          >
            <ForwardRef>
              Step 
              4
            </ForwardRef>
          </ForwardRef>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "completed": "filled",
            "theme": Object {
              "color": "rgb(58, 67, 141)",
            },
          },
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-EHOje",
              "isStatic": false,
              "rules": Array [
                "
  margin: 14px -5px 0 -5px;
  position: relative;
  height: 6px;
  width: 200px;
  border: 0.1em solid ",
                [Function],
                ";
  z-index: 0;
  ",
                [Function],
                ";
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-EHOje",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <ForwardRef>
              Step 
              4
            </ForwardRef>,
            "completed": "filled",
            "theme": Object {
              "color": "rgb(58, 67, 141)",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                "Step ",
                4,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Step ",
              4,
            ],
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-htoDjs",
                "isStatic": true,
                "rules": Array [
                  "
  font-size: 15px;
  margin-top: 40px;
  text-align: center;
  font-family: 'Courier New', Courier, monospace;
",
                ],
              },
              "displayName": "styled.p",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-htoDjs",
              "target": "p",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bZQynM",
              "isStatic": false,
              "rules": Array [
                "
  font-size: 60px;
  display: flex;
  width: 0.4em;
  box-sizing: content-box;
  height: 0.4em;
  border: 0.1em solid ",
                [Function],
                ";
  position: relative;
  border-radius: 0.35em;
  z-index: 1
  ",
                [Function],
                ";
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bZQynM",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
